packages/playwright/src/common/esmLoaderHost.ts

```diff
@@ -17,9 +17,21 @@
 import { addToCompilationCache, serializeCompilationCache } from '../transform/compilationCache';
 import { transformConfig } from '../transform/transform';
 import { PortTransport } from '../transform/portTransport';
+import url from 'url';
 
-const port = (globalThis as any).__esmLoaderPort;
-
+let port: MessagePort | undefined;
+if ((globalThis as any).__esmLoaderPort) {
+  port = (globalThis as any).__esmLoaderPort;
+  delete (globalThis as any).__esmLoaderPort;
+} else if (process.env.PW_TS_ESM_ON) {
+  const { port1, port2 } = new MessageChannel();
+  require('node:module').register(require.resolve('../transform/esmLoader'), {
+    parentURL: url.pathToFileURL(__filename),
+    data: { port: port2, from: 'playwright-test' },
+    transferList: [port2],
+  });
+  port = port1;
```

Comment:
As discussed, it seems that we don't need to restart any more but may need to wait for a response from our esm loader in case of Node 21+.